### ISSUE TITLE
ETAs for training?

### ISSUE THREAD

**Issue by IngwiePhoenix** [jaywalnut310/vits#223]
Hello there!

I am trying to work on a solution for a blind friend of mine to make all non-spoken texts in Persona 4 and 5 available to him since most OCR software does not seem to handle the in-game text very well. The idea is to train off of the available in-game samples and produce voices that can be used to produce audio for those textboxes that do not have voice data.

So, this is effectively a two-step process: Train a voice off of the voice clips of a character, and then run inference on all the texts that need a voice. However, there's like...a good dozen characters.

How long would it take to train, and is there a "checkpoint" system? As in, if I let it train for N hours a day and then stop, and resume the next, would that be possible? Because I am verry much sure that a single 4090 won't train a voice within one mere day.

Thank you very much and kind regards,
Ingwie

**Comment by piyushsinghpasi** [comment]
Hii I have trained the model, it usually takes 2-3 days to reach good quality (when trained from scratch). The paper uses ~24hr single speaker dataset (LJSpeech) and ~40hr (total) multispeaker dataset (VCTK). But the time estimate is on 1-2 A100 GPU (80GB). Not sure if single 4090 (24GB) would suffice, training was expensive. Maybe you can try smaller batch size and gradient accumulation to save memory, and finetuning existing checkpoints for faster convergence (given it's English language just different speaker).

**Comment by IngwiePhoenix** [comment]
That's genuenly shorter than I would've thought. Thank you for the detailed reply!

Yeah, just one 4090 sounds tough if you needed 1-2 A100s... Considering there has been a bit of an "update" to how model training can be done (see the Deepseek releases), I wonder if there's something I can leverage there - although that is LLM, not for voice.

Welp, I better get down to organizing all the voice samples into LJSpeech structure then - sounds like that's something I can prepare either way.

Since this very much answered my question, I am going to close the ticket, too. Again, thanks a lot!
